src/PkgServer.jl

```diff
@@ -90,13 +90,28 @@ function start(;kwargs...)
     update_registries()
 
     @sync begin
-        global last_registry_update
-        global registry_task = @spawn while true
-            last_registry_update = now()
-            sleep(1)
-            @try_printerror begin
-                forget_failures()
-                update_registries()
+        global last_registry_update = now()
+        global registry_update_task = @spawn begin
+            while true
+                last_registry_update = now()
+                sleep(1)
+                @try_printerror begin
+                    forget_failures()
+                    update_registries()
+                end
+            end
+        end
+
+        # Registry watchdog; if `last_registry_update` doesn't change
+        # for more than 20 minutes, we kill ourselves so that we can restart
+        global registry_watchdog_task = @spawn begin
+            while istaskstarted(registry_update_task)
+                sleep(30)
+                time_lag = now() - last_registry_update
+                if time_lag > 20*60
+                    @error("registry update watchdog timer tripped: $(time_lag) > $(20*60)")
```

Comment:
Should we try to throw an InterruptException to the update task to obtain the error here?

Reply:
Hmmm, that's a good idea!

Reply:
looks like that's not possible.  Jameson assures me that `Base.throwto()` doesn't actually do what we want it to, and that is reinforced by this error:

```
julia> t = @async sleep(10)
       Base.throwto(t, InterruptException())
       fetch(t)

WARNING: Workqueue inconsistency detected: popfirst!(Workqueue).state != :runnable
```

Reply:
I was thinking of
```
julia> t = @async sleep(10)
       schedule(t, InterruptException(); error=true)
       fetch(t)
ERROR: TaskFailedException
Stacktrace:
 [1] wait
   @ ./task.jl:317 [inlined]
 [2] fetch(t::Task)
   @ Base ./task.jl:332
 [3] top-level scope
   @ REPL[3]:3

    nested task error: InterruptException:
```